elastic/kibana
# [Dashboard navigation] Inject/extract references

Fixes https://github.com/elastic/kibana/issues/154363

## Summary

Extracts and injects references to Dashboards in Link panels to allow sharing / importing to other Spaces.

To test this:

1. Create an empty dashboard.
1. Create a Links panel on the dashboard and save to library.
1. Save the dashboard as one or more new dashboards.
1. Update the Links panel with links to the new dashboards.
1. Create two new Spaces, "space1" and "space2" with the default settings.
1. In Saved Objects, click the Actions button for your Links and choose "Copy to Spaces".
1. Copy the Links to "space1".
1. Switch to the "space1" space and verify all the dashboards using that Links panel were included.
1. In Saved Objects, select the Links saved object and export it including all related objects
1. Switch to "space2" and import the Links saved object from the file downloaded in the previous step.
1. Verify all the dashboards were also imported.
1. Inspect the saved object for the Links. The `references` array should contain objects for each dashboard.
1. The `attributes.links[].destination` property should be a string in the format of `links_<someuuid>_dashboard`. This string should match one of the `references[].id`.
